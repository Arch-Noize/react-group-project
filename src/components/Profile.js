import React from 'react';
import { useSelector } from 'react-redux';
import Table from 'react-bootstrap/Table';

const Profile = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const resRockets = rockets.filter((rocket) => rocket.reserved);

  const { dragons } = useSelector((state) => state.dragons);
  const resDragons = dragons.filter((dragon) => dragon.reserved);

  const { missions } = useSelector((state) => state.missions);
  const resMissions = missions.filter((mission) => mission.reserved);

  const renderReservedItemsTable = (items, title) => (
    <div className="table">
      <h2>{title}</h2>
      <Table bordered hover>
        <tbody>
          {items.map((item, index) => (
            <tr key={item.id}>
              <td>{item.name}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );

  return (
    <div className="profile">
      <div className="table-container">
        {renderReservedItemsTable(resRockets, 'Reserved Rockets')}
        {renderReservedItemsTable(resDragons, 'Reserved Dragons')}
        {renderReservedItemsTable(resMissions, 'Reserved Missions')}
      </div>
    </div>
  );
};

export default Profile;
