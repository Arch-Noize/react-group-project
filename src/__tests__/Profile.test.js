import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Profile from '../components/Profile';

const mockStore = configureMockStore([]);

describe('Loading profile component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      rockets: {
        rockets: [],
      },
      dragons: {
        dragons: [],
      },
      missions: {
        missions: [],
      },
    });

    const { tree } = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
