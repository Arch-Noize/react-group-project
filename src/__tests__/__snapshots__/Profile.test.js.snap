// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading profile component renders correctly 1`] = `undefined`;
